src/Workspaces/MSBuild/BuildHost/MSBuild/ProjectFile/ProjectInstanceReader.cs

```diff
@@ -0,0 +1,256 @@
+            AnalyzerConfigDocuments = analyzerConfigDocs,
+            ProjectReferences = [.. _projectInstance.GetProjectReferences()],
+            PackageReferences = packageReferences,
+            ProjectCapabilities = projectCapabilities,
+            ContentFilePaths = contentFileInfo,
+            FileGlobs = fileGlobs
+        };
+
+        static FileGlobs GetFileGlobs(MSB.Evaluation.GlobResult g)
+        {
+            return new FileGlobs(
+                Includes: [.. g.IncludeGlobs.Select(PathUtilities.ExpandAbsolutePathWithRelativeParts)],
+                Excludes: [.. g.Excludes.Select(PathUtilities.ExpandAbsolutePathWithRelativeParts)],
+                Removes: [.. g.Removes.Select(PathUtilities.ExpandAbsolutePathWithRelativeParts)]);
+        }
+    }
+
+    private static ImmutableArray<string> GetContentFiles(MSB.Execution.ProjectInstance project)
+    {
+        var contentFiles = project
+            .GetItems(ItemNames.Content)
+            .SelectAsArray(item => item.GetMetadataValue(MetadataNames.FullPath));
+        return contentFiles;
+    }
+
+    private ImmutableArray<string> GetCommandLineArgs(MSB.Execution.ProjectInstance project)
+    {
+        var commandLineArgs = _commandLineProvider.GetCompilerCommandLineArgs(project)
+            .SelectAsArray(item => item.ItemSpec);
+
+        if (commandLineArgs.Length == 0)
+        {
+            // We didn't get any command-line args, which likely means that the build
+            // was not successful. In that case, try to read the command-line args from
+            // the ProjectInstance that we have. This is a best effort to provide something
+            // meaningful for the user, though it will likely be incomplete.
+            commandLineArgs = _commandLineProvider.ReadCommandLineArgs(project);
+        }
+
+        return commandLineArgs;
+    }
+
+    private static bool IsNotTemporaryGeneratedFile(MSB.Framework.ITaskItem item)
+        => !Path.GetFileName(item.ItemSpec).StartsWith("TemporaryGeneratedFile_", StringComparison.Ordinal);
+
+    private DocumentFileInfo MakeDocumentFileInfo(MSB.Framework.ITaskItem documentItem)
+    {
+        var filePath = GetDocumentFilePath(documentItem);
+        var logicalPath = GetDocumentLogicalPath(documentItem, _projectDirectory);
+        var isLinked = IsDocumentLinked(documentItem);
+
+        var folders = GetRelativeFolders(documentItem);
+        return new DocumentFileInfo(filePath, logicalPath, isLinked, isGenerated: false, folders);
+    }
+
+    private DocumentFileInfo MakeNonSourceFileDocumentFileInfo(MSB.Framework.ITaskItem documentItem)
+    {
+        var filePath = GetDocumentFilePath(documentItem);
+        var logicalPath = GetDocumentLogicalPath(documentItem, _projectDirectory);
+        var isLinked = IsDocumentLinked(documentItem);
+
+        var folders = GetRelativeFolders(documentItem);
+        return new DocumentFileInfo(filePath, logicalPath, isLinked, isGenerated: false, folders);
```

Comment:
This also used to check existence of Compile item in the Project file, which is entirely wrong since Compile items have nothing to do with additional or editor config files.